cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/template/etl/batch/BaseETLBatchTest.java

```diff
@@ -70,4 +78,29 @@ public static void setupTest() throws IOException {
       PipelineConfigurable.class.getPackage().getName(),
       BatchSource.class.getPackage().getName());
   }
+
+  protected List<GenericRecord> readOutput(TimePartitionedFileSet fileSet, Schema schema) throws IOException {
+    org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(schema.toString());
+    DatumReader<GenericRecord> datumReader = new GenericDatumReader<>(avroSchema);
+    List<GenericRecord> records = com.google.common.collect.Lists.newArrayList();
+    for (Location dayLoc : fileSet.getEmbeddedFileSet().getBaseLocation().list()) {
+      // this level should be the day (ex: 2015-01-19)
+      for (Location timeLoc : dayLoc.list()) {
+        // this level should be the time (ex: 21-23.1234567890000)
+        for (Location file : timeLoc.list()) {
+          // this level should be the actual mapred output
+          String locName = file.getName();
+
+          if (locName.endsWith(".avro")) {
+            DataFileStream<GenericRecord> fileStream =
+              new DataFileStream<>(file.getInputStream(), datumReader);
+            while (fileStream.hasNext()) {
```

Comment:
`DataFileStream` implements `Iterable`. You can simply do `Iterables.addAll(records, fileStream)`. Also, you need to close the stream.

```java
try (DataFileStream<GenericRecord> fs = new ...) {
  Iterables.addAll(records, fs);
}
```